addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -215,15 +215,28 @@ function factory(dependencies) {
             } else if (!this.mimetype) {
                 return clear();
             }
+            switch (this.mimetype) {
+                case 'image/bmp':
+                case 'image/gif':
+                case 'image/jpeg':
+                case 'image/png':
+                case 'image/webp':
+                    return 'image';
+                case 'text/css':
+                case 'text/html':
+                case 'text/plain':
+                    return 'text';
+                case 'video/mp4':
+                case 'video/ogg':
+                case 'video/webm':
+                    return 'video';
+            }
             const match = this.type === 'url'
                 ? this.url.match('(youtu|.png|.jpg|.gif)')
-                : this.mimetype.match('(image|video|application/pdf|text)');
+                : this.mimetype.match('(application/pdf)');
```

Comment:
There should be no check on mimetype here, you already made the switch above.
It's either mimetype or url, I don't understand why the 2 are mixed up here.